staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles.go

```diff
@@ -49,11 +61,167 @@ func profileLegacy(pod *corev1.Pod, containerName string, target runtime.Object)
 			})
 		}
 
-		pod.Spec.HostIPC = true
 		pod.Spec.HostNetwork = true
 		pod.Spec.HostPID = true
+		pod.Spec.HostIPC = true
 		return nil
 	default:
 		return fmt.Errorf("the %s profile doesn't support objects of type %T", ProfileLegacy, target)
 	}
 }
+
+func (p *generalProfile) Apply(pod *corev1.Pod, containerName string, target runtime.Object) error {
+	switch t := target.(type) {
+	case *corev1.Pod:
+		if t == pod {
+			// For ephemeral container: sets SYS_PTRACE in ephemeral container
+			for i, c := range pod.Spec.EphemeralContainers {
+				if c.Name == containerName {
+					pod.Spec.EphemeralContainers[i].SecurityContext = &corev1.SecurityContext{
+						Capabilities: &corev1.Capabilities{
+							Add: []corev1.Capability{"SYS_PTRACE"},
+						},
+					}
+					if c.SecurityContext == nil {
+					} else if c.SecurityContext.Capabilities == nil {
+						pod.Spec.EphemeralContainers[i].SecurityContext.Capabilities = &corev1.Capabilities{
+							Add: []corev1.Capability{"SYS_PTRACE"},
+						}
+					} else {
+						pod.Spec.EphemeralContainers[i].SecurityContext.Capabilities.Add = append(c.SecurityContext.Capabilities.Add, "SYS_PTRACE")
+					}
+				}
+			}
+		} else {
+			// For copy of pod: sets SYS_PTRACE in debugging container, sets shareProcessNamespace
+			pod.Spec.ShareProcessNamespace = pointer.BoolPtr(true)
+			for i, c := range pod.Spec.Containers {
```

Comment:
Maybe to get rid of `else if` statements, we can change this to something like tha?;

```go
				if c.Name != containerName {
					continue
				}
				
				pod.Spec.Containers[i].LivenessProbe = nil
				pod.Spec.Containers[i].ReadinessProbe = nil
				
				if c.SecurityContext == nil {
					pod.Spec.Containers[i].SecurityContext = &corev1.SecurityContext{
						Capabilities: &corev1.Capabilities{
							Add: []corev1.Capability{"SYS_PTRACE"},
						},
					}
					continue
				}
				
				if c.SecurityContext.Capabilities == nil {
					pod.Spec.Containers[i].SecurityContext.Capabilities = &corev1.Capabilities{
						Add: []corev1.Capability{"SYS_PTRACE"},
					}
					continue
				}
				
				pod.Spec.Containers[i].SecurityContext.Capabilities.Add = append(c.SecurityContext.Capabilities.Add, "SYS_PTRACE")
```

Reply:
But I'd like to note that, code snippet above clears probes only for debug container and not others. If we need to clear probes for all containers. We need to change that.